Server/JokesOnYou.Web.Api/Services/UserService.cs

```diff
@@ -1,39 +1,60 @@
 ﻿using JokesOnYou.Web.Api.DTOs;
 using JokesOnYou.Web.Api.Models;
+using JokesOnYou.Web.Api.Repositories;
 using JokesOnYou.Web.Api.Repositories.Interfaces;
 using JokesOnYou.Web.Api.Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Components;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using JokesOnYou.Web.Api.Exceptions;
+using AutoMapper;
 
 namespace JokesOnYou.Web.Api.Services
 {
+    
     public class UserService : IUserService
     {
-        private readonly IUserRepository _userRepository;
-
-        public UserService(IUserRepository userRepository)
+        readonly IUserRepository _userRepo;
+        readonly IMapper _mapper;
+        public  UserService(IUserRepository userRepo, IMapper mapper)
         {
-            _userRepository = userRepository;
+            _userRepo = userRepo;
+            _mapper = mapper;
         }
-
+        public async Task<IEnumerable<UserReplyDTO>> GetAll()
+        {
+             var users = await _userRepo.GetUsersAsync();
+            return _mapper.Map<List<UserReplyDTO>>(users);
+        }
+        /*
+        public async Task<User> CreateUser(UserRegisterDTO registerDTO)
+        {
+            return await _userRepo.CreateUserAsync(registerDTO);
+        }
+        */
         public async Task DeleteUser(string id)
         {
-            throw new NotImplementedException();
+            var user = await _userRepo.GetUserAsync(id);
+            if (user == null)
+                throw new AppException($"Cant find user of id:{id}");
+            await _userRepo.DeleteUserAsync(user);
         }
 
-        public async Task<IEnumerable<User>> GetAll()
+        public async Task<UserReplyDTO> GetUserReplyById(string id)
         {
-            throw new NotImplementedException();
+            return await _userRepo.GetUserReplyAsync(id);
         }
 
-        public async Task<User> GetUserById(string id)
+        public async Task UpdateUser(UserUpdateDTO userDTO)
         {
-            throw new NotImplementedException();
+            var user = _mapper.Map<User>(userDTO);
```

Comment:
this will create a new user in stead of updating an existing one.

Reply:
am I wrong here @chrisK00 ?
or if an id is given it will map to the correct user ?

Reply:
First is correct, automapper does not know about ids etc. just namings. 

Current one: Map<toNewEntity>(fromExistingEntity) 
How it should be: Map(fromExistingEntity, ToExistingEntity)